recipes-support/snapd/snapd-go.inc

```diff
@@ -0,0 +1,64 @@
+inherit go-mod
+
+GO_IMPORT = "github.com/snapcore/snapd"
+
+# disable shared runtime for x86
+# https://forum.snapcraft.io/t/yocto-rocko-core-snap-panic/3261
+# GO_DYNLINK is set with arch overrides in goarch.bbclass
+GO_DYNLINK:x86 = ""
+GO_DYNLINK:x86-64 = ""
+GO_DYNLINK:arm = ""
+GO_DYNLINK:aarch64 = ""
+# modules are vendored in the release tarball
+GOBUILDFLAGS:append = " -mod=vendor"
+
+GO_BUILD_TAGS_snapd = "nosecboot"
+GO_BUILD_TAGS = "nosecboot nomanagers"
+
+# The go class does export a do_configure function, of which we need
+# to change the symlink set-up, to target snapd's environment.
+snapd_go_do_configure() {
+	mkdir -p ${S}/src/github.com/snapcore
+	ln -snf ${S} ${S}/src/${GO_IMPORT}
+	go_do_configure
+	# internally calls go run to generate some assets
+	(cd ${S} ; GOARCH=${GOHOSTARCH} sh -x ./mkversion.sh ${PV})
+}
+
+snapd_go_do_compile() {
+	(
+		cd ${S}
+		${GO} install -tags '${GO_BUILD_TAGS_snapd}' ${GOBUILDFLAGS} github.com/snapcore/snapd/cmd/snapd
```

Comment:
Perhaps add a comment why GO_BUILD_TAGS are single-quoted everywhere, it looks puzzling coming from pure-shell perspective.